### PR TITLE
[libical] new port

### PR DESCRIPTION
**Pull request by wrobelda**
**Describe the pull request**
  Adds new LibIcal library. Builds on top of #8754 

  Depends on:
 - [x] #20966 

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  All official, plus `arm64-osx`

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/libical/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    list(APPEND FEATURE_OPTIONS -DSTATIC_ONLY=ON)
+else()
+    list(APPEND FEATURE_OPTIONS -DSHARED_ONLY=ON)
+endif()
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DCMAKE_DISABLE_FIND_PACKAGE_BDB=ON
+        -DUSE_BUILTIN_TZDATA=ON
```
[the remark this answers is not in the document]
Comment (winterz) [reply ports/libical/portfile.cmake:27]: fixed . DEF_USE_BUILTIN_TZDATA is now True for Windows and False on non-Windows
this is in the master branch.  the stable 3.0 branch is correct as is.

File: ports/libical/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        -DICAL_GLIB=OFF
```
[the remark this answers is not in the document]
Comment (wrobelda) [reply ports/libical/portfile.cmake:28]: @winterz tried adding "glib" feature, but getting:
```
CMake Error:
  Running

   '/Users/cromo/Documents/Sourcecode/vcpkg/downloads/tools/ninja-1.10.2-osx/ninja' '-C' '/Users/cromo/Documents/Sourcecode/vcpkg/buildtrees/libical/arm64-osx-dbg' '-t' 'recompact'

  failed with:

   ninja: error: build.ninja:2151: multiple rules generate lib/libical-glib.a [-w dupbuild=err]
```

**Comment by wrobelda** [comment]
@PhoebeHui this is ready to build now.

**Review by PhoebeHui — APPROVED** [review]
LGTM. BTW, since the feature 'rscale' has been restricted to non-static builds, the https://github.com/microsoft/vcpkg/issues/20966 will not block this PR.

**Pull request merged**